src/shims/unix/fd.rs

```diff
@@ -187,28 +188,35 @@ impl FileDescription for NullOutput {
         Ok(Ok(bytes.len()))
     }
 }
+/// Structure contains both the file description and its unique identifier.
+#[derive(Clone, Debug)]
+pub struct FileDescWithID<T: FileDescription + ?Sized> {
+    #[allow(dead_code)]
+    id: usize,
+    file_description: RefCell<Box<T>>,
+}
```

Comment:
Similary, this one is not needed here

Reply:
You mean the ``id`` field or the entire ``FileDescWithID`` struct? I currently only remove the ``id`` field but let me know if you mean the latter.